lib/paths/index.ts

```diff
@@ -20,25 +21,33 @@ type DefaultRecursivePathsOptions = {
  */
 type RecursivePathsOptions = {
   depth: any[];
+  anyArrayIndexAccessor: string | false;
```

Comment:
`RecursivePathsOptions` is required for `CallOptions`. Since you don't accumulate anything during recursive calls (unlike `depth` where type will contain the number of recursive calls), I suggest removing the value here